mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -453,11 +460,64 @@ public boolean removeClanAsEnemy(Transaction transaction, Clan clan) {
      * @return All clans this clan has currently marked as enemies
      */
     public Set<Clan> getEnemyClans(Transaction transaction) {
-        return transaction.getProperty(DSLContext.class)
+        var result = transaction.getProperty(DSLContext.class)
                 .select(CLANS_CLAN_ENEMIES.ENEMY_ID)
                 .from(CLANS_CLAN_ENEMIES)
                 .where(CLANS_CLAN_ENEMIES.CLAN_ID.eq(this.clanId))
-                .fetchSet((record) -> manager.getClan(transaction, record.get(CLANS_CLAN_ENEMIES.ENEMY_ID)));
+                .fetch();
+
+        Set<Clan> clans = new HashSet<>();
+
+        result.forEach(res -> {
+            Optional<Clan> op = manager.getClanById(transaction, res.get(CLANS_CLAN_ENEMIES.ENEMY_ID));
+
+            op.ifPresent(clans::add);
+        });
+
+        return clans;
+    }
+
+
+    /**
+     * Sets the owner of the clan to the user provided by the id, and
+     * if they are not a member of the clan they are added.
+     *
+     * The old owner remains a member of the clan.
+     *
+     * @param transaction tx
+     * @param id user id of the new clan owner
+     */
+    public void setLeader(Transaction transaction, SolarPlayer id) {
+
+        var context = transaction.getProperty(DSLContext.class);
+
+        context
+                .update(CLANS_CLAN_INFO)
+                .set(CLANS_CLAN_INFO.CLAN_LEADER, id.getUserId())
+                .where(CLANS_CLAN_INFO.CLAN_ID.eq(this.clanId))
+                .execute();
+
+        int result = context.select(clansAddMember(this.clanId, id.getUserId()))
+                .execute();
+
+        this.leader = new ClanMember(id.getUserId());
+
+        //if user is already a member do nothing since the documentation says so
+
+    }
+
+    /**
+     * Sets the name of the clan
+     *
+     * @param transaction transaction
+     * @param name name
+     */
+    public void setName(Transaction transaction, String name){
+        transaction.getProperty(DSLContext.class)
```

Comment:
You need to update the `clanName` field in this implementation.

You should write an integration test for this method as well as setLeader. Tests aren't hard, they will take a bit of time, but they will be faster than asking me for another review, and they will help you uncover bugs.

Reply:
ok

Reply:
literally just copy and pasted what zone wrote